bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/v18/BitcoindV18RpcClient.scala

```diff
@@ -0,0 +1,45 @@
+package org.bitcoins.rpc.client.v18
+import akka.actor.ActorSystem
+import org.bitcoins.core.crypto.ECPrivateKey
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.rpc.client.common.{
+  BitcoindRpcClient,
+  BitcoindVersion,
+  RpcOpts
+}
+import org.bitcoins.rpc.config.BitcoindInstance
+import org.bitcoins.rpc.jsonmodels.{
+  AddressInfoResult,
+  SignRawTransactionResult,
+  TestMempoolAcceptResult
+}
+import org.bitcoins.rpc.serializers.JsonSerializers._
+import org.bitcoins.rpc.serializers.JsonWriters._
+import play.api.libs.json.{JsArray, JsBoolean, JsString, Json}
+
+import scala.concurrent.Future
+import scala.util.Try
+
+class BitcoindV18RpcClient(override val instance: BitcoindInstance)(
+  implicit
+  actorSystem: ActorSystem)
+  extends BitcoindRpcClient(instance)
+    with V18PsbtRpc {
+
+  override def version: BitcoindVersion = BitcoindVersion.V18
```

Comment:
this can be 

```scala
override val version: BitcoindVersion = BitcoindVersion.V18
```

`val` will _not_ re-evaluate the expression everytime it is called, it caches the value. If you use a `def` you re-evaluate the expression every single time. It is pointless to re-evaluated the expression everytime if it does not change.

Reply:
So this is actually not entirely correct, and having this as a `val` causes things to crash. This is due to some rather confusing things going on in initialization of traits in Scala. If this is a `val` we get this error: 
`  java.lang.IllegalArgumentException: requirement failed: bitcoind version must be null, got v0.18`

I.e. the version is null when we evaluate it, because of wonky initialization logic in traits and classes. 

`def` works, but so does `lazy val`. `lazy val` only gets evaluated once, so I'd say that's best to use.